manifests/generate.sh

```diff
@@ -135,19 +135,21 @@ ${HELM} template \
 	--set calicoctl.image=$REGISTRY/ctl \
 	--set calicoctl.tag=$CALICO_VERSION
 # The first two lines are a newline and a yaml separator - remove them.
-find ocp/tigera-operator -name "*.yaml" | xargs sed -i -e 1,2d
+find ocp/tigera-operator -name "*.yaml" -print0 | xargs -0 sed -i -e 1,2d
 mv $(find ocp/tigera-operator -name "*.yaml") ocp/ && rm -r ocp/tigera-operator
 
 # Generating the upgrade manifest for OCP.
-# It excludes the CRs (01-*) and the specific BPF files to maintain compatibility with iptables.
-VALUES_FILES=$(ls ocp | grep -v -e '^01-' -e 'cluster-network-operator.yaml' -e '02-configmap-calico-resources.yaml')
+# It excludes files specific to using BPF, CRs (03-cr-*) and CRDs to maintain compatibility with iptables.
+OCP_VALUES_FILES=$(ls ocp | grep -v -e '^01-' -e 'cluster-network-operator.yaml' -e '02-configmap-calico-resources.yaml' -e '^03-cr-' -e 'crd')
 rm -f tigera-operator-ocp-upgrade.yaml
-for FILE in $VALUES_FILES; do
+for FILE in $OCP_VALUES_FILES; do
   cat "ocp/$FILE" >> tigera-operator-ocp-upgrade.yaml
-  echo -e "---" >> tigera-operator-ocp-upgrade.yaml  # Add divisor
+  echo -e "---" >> tigera-operator-ocp-upgrade.yaml  # Add separator
 done
-# Remove the last separator (last line)
-sed -i -e '$ d' tigera-operator-ocp-upgrade.yaml
+# Remove the last separator
+sed -i -e '${/^---$/d}' tigera-operator-ocp-upgrade.yaml
```

Comment:
Is this just cosmetic? or actually important?

Reply:
again, I was just trying to make the `sed` expression a little more robust, but yes I do believe it's not invalid yaml, so yes cosmetic. should we leave the last `---` separator?

Reply:
It was cosmetic, I tested. Removed this to keep the script simpler